capsule.sh

```diff
@@ -0,0 +1,267 @@
+    local edk_tools=${edk_workspace}/Dasharo/BaseTools/BinWrappers/PosixLike
+
+    if [ ! -f .config ]; then
+        die "no '.config' file in current directory"
+    fi
+    if [ ! -f build/coreboot.rom ]; then
+        die "no 'build/coreboot.rom'; the firmware wasn't built?"
+    fi
+    if [ ! -x "${edk_tools}/GenerateCapsule" ]; then
+        die "'${edk_tools}/GenerateCapsule' can't be executed"
+    fi
+
+    # import coreboot's config file replacing $(...) with ${...}
+    while read -r line; do
+        if ! eval "$line"; then
+            die "failed to source '.config'"
+        fi
+    done <<< "$(sed 's/\$(\([^)]\+\))/${\1}/g' .config)"
+
+    if [ "$CONFIG_DRIVERS_EFI_UPDATE_CAPSULES" != y ]; then
+        die "current board configuration lacks support of update capsules"
+    fi
+
+    # Option names match terminology of GenerateCapsule which conveniently start
+    # with different letters:
+    #  * t - trusted
+    #  * o - other
+    #  * s - signer
+    local root_cert sub_cert sign_cert
+    while getopts "t:o:s:" OPTION; do
+        case $OPTION in
+            t) root_cert="$OPTARG" ;;
+            o) sub_cert="$OPTARG" ;;
+            s) sign_cert="$OPTARG" ;;
+            *) exit 1 ;;
+        esac
+    done
+
+    check_cert root "$root_cert"
+    check_cert sub "$sub_cert"
+    check_cert sign "$sign_cert"
+
+    local cap_file=${CONFIG_MAINBOARD_DIR/\//-}
+    if [[ ${CONFIG_MAINBOARD_PART_NUMBER} =~ DDR4 ]]; then
+        cap_file+=-ddr4
+    fi
+    cap_file+=-${CONFIG_LOCALVERSION}
+    cap_file+=.cap
+
+    if [ -e "$cap_file" ]; then
+        confirm "Overwrite already existing '$cap_file'?"
+    fi
+
+    local build_type
+    if [ "$CONFIG_EDK2_RELEASE" = y ]; then
+        build_type=RELEASE
+    else
+        build_type=DEBUG
+    fi
+
+    local json_file
+    json_file=$(mktemp --tmpdir --suffix -cap.json XXXXXXXX)
+    trap "$(printf 'rm -f %q' "$json_file")" EXIT
+
+    cat > "$json_file" << EOF
+{
+    "EmbeddedDrivers": [
+        {
+            "Driver": "${edk_workspace}/Build/DasharoPayloadPkgX64/${build_type}_COREBOOT/X64/FmpDxe.efi"
+        }
+    ],
+    "Payloads": [
+        {
+            "Payload": "build/coreboot.rom",
+            "Guid": "${CONFIG_DRIVERS_EFI_MAIN_FW_GUID}",
+            "FwVersion": "${CONFIG_DRIVERS_EFI_MAIN_FW_VERSION}",
+            "LowestSupportedVersion": "${CONFIG_DRIVERS_EFI_MAIN_FW_LSV}",
+
+            "OpenSslSignerPrivateCertFile": "${sign_cert}",
+            "OpenSslOtherPublicCertFile": "${sub_cert}",
+            "OpenSslTrustedPublicCertFile": "${root_cert}"
+        }
+    ]
+}
+EOF
+
+    # Linux doesn't support InitiateReset flag, omitting it to rely on manual
+    # warm reset
+    "${edk_tools}/GenerateCapsule" --encode \
+                                   --capflag PersistAcrossReset \
+                                   --json-file "$json_file" \
+                                   --output "$cap_file"
+    if [ $? -ne 0 ]; then
```

Comment:
Right, due to `set -e`.  Moved it into the conditional: https://github.com/Dasharo/coreboot/compare/cc44feab6eeff600804cfb7f397ad8a3acabdaf8..b93dedf8328a7af8d2db52a6f1354d5be3d352dd#diff-6bc34407b9d3bc8bc5bba23214f26c3a17237a614cd513cddc6d746a84551f46R242-R245 as shellcheck suggested anyway and I thought it might be less readable but it's fine either way.